src/inlay-hints.ts

```diff
@@ -212,15 +213,21 @@ class InlayHintsFeature implements vscodelc.StaticFeature {
   private hintsToDecorations(hints: InlayHint[]): InlayDecorations {
     const decorations: InlayDecorations = {parameterHints: [], typeHints: []};
     const conv = this.context.client.protocol2CodeConverter;
+    const showParameterHints = config.get<boolean>('inlayHints.parameterHints');
+    const showTypeHints = config.get<boolean>('inlayHints.typeHints');
     for (const hint of hints) {
       switch (hint.kind) {
       case InlayHintKind.Parameter: {
-        decorations.parameterHints.push(
-            parameterHintStyle.toDecoration(hint, conv));
+        if (showParameterHints) {
```

Comment:
nit: rather than branching for every hint, we can just clear parameterHints/typeHints before returning decorations at the bottom.